fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -180,9 +180,132 @@ def additional_upload_parameters
 
+  # Generates commands and executes the altool.
+  class AltoolTransporterExecutor < TransporterExecutor
+    ERROR_REGEX = /.*\*\*\*\sError:\s+(.+)/
+
+    private_constant :ERROR_REGEX
+
+    def execute(command, hide_output)
+      if Helper.test?
+        yield(nil) if block_given?
+        return command
+      end
+
+      @errors = []
+      @warnings = []
+      @all_lines = []
+
+      if hide_output
+        # Show a one time message instead
+        UI.success("Waiting for App Store Connect transporter to be finished.")
+        UI.success("Application Loader progress... this might take a few minutes...")
+      end
+
+      begin
+        exit_status = FastlaneCore::FastlanePty.spawn(command) do |command_stdout, command_stdin, pid|
+          begin
+            command_stdout.each do |line|
+              @all_lines << line
+              parse_line(line, hide_output) # this is where the parsing happens
+            end
+          end
+        end
+      rescue => ex
+        # FastlanePty adds exit_status on to StandardError so every error will have a status code
+        exit_status = ex.exit_status
+        @errors << ex.to_s
+      end
+
+      unless exit_status.zero?
+        @errors << "The call to the altool completed with a non-zero exit status: #{exit_status}. This indicates a failure."
+      end
+
+      if @warnings.count > 0
+        UI.important(@warnings.join("\n"))
+      end
+
+      if @errors.count > 0 && @all_lines.count > 0
+        # Print out the last 18 lines, this is key for non-verbose mode
+        @all_lines.last(18).each do |line|
+          UI.important("[altool] #{line}")
+        end
+        UI.message("Application Loader output above ^")
+        UI.error(@errors.join("\n"))
+      end
+
+      yield(@all_lines) if block_given?
+      return exit_status.zero?
+    end
+
+    def build_upload_command(username, password, source = "/tmp", provider_short_name = "", jwt = nil, platform = nil)
+      use_jwt = !jwt.to_s.empty?
+      [
+        "xcrun altool",
+        "--upload-app",
+        ("-u #{username.shellescape}" unless use_jwt),
+        ("-p #{password.shellescape}" unless use_jwt),
+        ("--appKey #{jwt}" if use_jwt),
+        "-t #{platform}",
+        file_upload_option(source),
+        additional_upload_parameters,
+        "-k 100000"
+      ].compact.join(' ')
+    end
+
+    def additional_upload_parameters
+      env_deliver_additional_params = ENV["DELIVER_ALTOOL_ADDITIONAL_UPLOAD_PARAMETERS"]
+      if env_deliver_additional_params.to_s.strip.empty?
+        return nil
+      end
+
+      return env_deliver_additional_params.to_s.strip
+    end
+
+    def file_upload_option(source)
+      return "-f #{source.shellescape}"
+    end
+
+    def handle_error(password)
+      UI.error("Could not download/upload from App Store Connect!")
+    end
+
+    def displayable_errors
+      @errors.map { |error| "[Application Loader Error Output]: #{error}" }.join("\n").gsub!(/"/, "")
```

Comment:
Use `message` to display.
https://ruby-doc.org/core-2.6/Exception.html#method-i-message

Reply:
this `error` is string which is obtained from output from `altool`, not Exception.